.github/workflows/ci.yml

```diff
@@ -0,0 +1,44 @@
+name: CI Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build_lint_test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Setup Mage
+        uses: magefile/mage-action@v3
+        with:
+          install-only: true
+
+      - name: Install test dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y gcc
+
+      - name: Setup CI Linter
+        run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.54.2
```

Comment:
Nit: In order to make it easier to bump golangci-lint in the future, I'd move the version to an environment variable at the top of the file:

```yml
env:
  GOLANGCI_LINT_VERSION: v1.54.2

# ...
jobs:
  # ...
  build_lint_test:
    # ...
    steps:
      - name: Setup CI Linter
        run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin ${{ env.GOLANGCI_LINT_VERSION }}
```